core/store/migrations/0041_create_job_proposals.go

```diff
@@ -0,0 +1,42 @@
+package migrations
+
+import (
+	"gorm.io/gorm"
+)
+
+const up41 = `
+CREATE TYPE job_proposal_status AS ENUM ('pending', 'approved', 'rejected');
+CREATE TABLE job_proposals (
+	id BIGSERIAL PRIMARY KEY,
+	spec TEXT NOT NULL ,
+	status job_proposal_status NOT NULL,
+	job_id uuid REFERENCES jobs (external_job_id),
```

Comment:
Here you can choose between either `DEFERRABLE INITIALLY IMMEDIATE` or `DEFERRABLE INITIALLY DEFERRED`.

The second one might be nice because it would allow you to do this:

```
// start transaction
// update job_proposal set job_id = foo
// create job with external_job_id = foo
// commit
```